core/vm/contracts.go

```diff
@@ -669,3 +676,50 @@ func (c *epochSize) Run(input []byte, caller common.Address, evm *EVM, gas uint6
 
 	return epochSizeBytes, gas, nil
 }
+
+type blockNumberFromHeader struct{}
+
+func (c *blockNumberFromHeader) RequiredGas(input []byte) uint64 {
+	return params.GetBlockNumberFromHeaderGas
+}
+
+func (c *blockNumberFromHeader) Run(input []byte, caller common.Address, evm *EVM, gas uint64) ([]byte, uint64, error) {
+	gas, err := debitRequiredGas(c, input, gas)
+	if err != nil {
+		return nil, gas, err
+	}
+
+	var header types.Header
+	err = rlp.Decode(bytes.NewReader(input), &header)
```

Comment:
No need to create the bytes reader https://godoc.org/github.com/ethereum/go-ethereum/rlp#DecodeBytes
```suggestion
	err = rlp.DecodeBytes(input, &header)
```